samples/ar-view-node/src/main/java/io/github/sceneview/sample/arviewnode/MainActivity.kt

```diff
@@ -0,0 +1,162 @@
+package io.github.sceneview.sample.arviewnode
+
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import com.google.ar.core.Config
+import io.github.sceneview.ar.ARSceneView
+import io.github.sceneview.ar.arcore.createAnchorOrNull
+import io.github.sceneview.ar.node.AnchorNode
+import io.github.sceneview.sample.arviewnode.nodes.events.ImageNodeEvent
+import io.github.sceneview.sample.arviewnode.nodes.events.ViewNodeEvent
+import io.github.sceneview.sample.arviewnode.nodes.ImageNodeHelper
+import io.github.sceneview.sample.arviewnode.nodes.TrackingNode
+import io.github.sceneview.sample.arviewnode.nodes.ViewNodeHelper
+import io.github.sceneview.sample.arviewnode.utils.NodeRotationHelper
+
+
+class MainActivity : AppCompatActivity(), MainActivityViewMvc.Listener {
+
+    private lateinit var viewMvc: MainActivityViewMvc
+
+    private lateinit var trackingNode: TrackingNode
+    private lateinit var imageNodeHelper: ImageNodeHelper
+    private lateinit var viewNodeHelper: ViewNodeHelper
+    private lateinit var nodeRotationHelper: NodeRotationHelper
+
+    private val nodeList: MutableList<AnchorNode> = mutableListOf()
+
+    private var nodeType = NodeType.VIEW_NODE
+
+    ////////////////////
+    // region Lifecycle Methods
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        viewMvc = MainActivityViewMvcImpl(
+            this,
+            layoutInflater,
+            null
+        )
+        setContentView(viewMvc.getView().root)
+        viewMvc.registerListener(this)
+
+        viewMvc.getSceneView().apply {
+            initObjects(this)
+            configureArSession(this)
+            handleOnSessionUpdated(this)
+
+            lifecycle = this@MainActivity.lifecycle
+        }
+    }
+
+    private fun initObjects(arSceneView: ARSceneView) {
+        trackingNode = TrackingNode(
+            this@MainActivity,
+            arSceneView
+        )
+        imageNodeHelper = ImageNodeHelper(
+            this@MainActivity,
+            arSceneView,
+            this@MainActivity::onEvent
+        )
+        viewNodeHelper = ViewNodeHelper(
+            this@MainActivity,
+            arSceneView,
+            this@MainActivity::onEvent
+        )
```

Comment:
It seems like the class reference after `this` isn't required here.

Reply:
True a leftover after a refactoring step. I fix it

Reply:
fixed